### PR TITLE
Assign deep defaults for custom config

### PR DESCRIPTION
**Pull request by kinetifex**
* Fixes deep defaults not being included when using customConfig.
Carried over from https://github.com/zeit/next.js/pull/6305

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Could you run `yarn lint-fix` then I'll merge 🙏

**Comment by timneutkens** [comment]
Thanks!

**Comment by indexzero** [comment]
Thanks for the quick turn around @timneutkens!

File: packages/next-server/server/config.js [hunk 1]
```diff
@@ -25,6 +25,20 @@ const defaultConfig = {
   }
 }
 
+function assignDefaults (userConfig) {
+  Object.keys(userConfig).forEach(key => {
+    const maybeObject = userConfig[key]
+    if ((!!maybeObject) && (maybeObject.constructor === Object)) {
```
Comment (popuguytheparrot) [inline packages/next-server/server/config.js]: use Boolean(maybeObject)

**Pull request merged**